### PR TITLE
Spinnaker Application Enhancements (#7)

### PR DESCRIPTION
**Pull request by guido9j**
Enhancements to Spinnaker application resource:

- Add support for cloud providers
- Add support for permissions
- Implemented Update method
- Improved error messages for Spinnaker application

### CODE REVIEW

File: spinnaker/resource_application.go [hunk 1]
```diff
@@ -51,16 +87,18 @@ type applicationRead struct {
...
+func resourceApplicationCreateOrUpdate(data *schema.ResourceData, meta interface{}) error {
 	clientConfig := meta.(gateConfig)
 	client := clientConfig.client
 	application := data.Get("application").(string)
 	email := data.Get("email").(string)
 	description := data.Get("description").(string)
 	platform_health_only := data.Get("platform_health_only").(bool)
 	platform_health_only_show_override := data.Get("platform_health_only_show_override").(bool)
+	cloud_providers := data.Get("cloud_providers").([]interface {})
+	permissions := data.Get("permissions").(interface {})
```
Comment (pjberry16) [inline spinnaker/resource_application.go:99]: You're asserting the type as interface{} here then *schema.Set on line 101 is there any reason you don't just do something like this here?:
```
if permissions, ok := data.Get("permissions").(*schema.Set); !ok {
  some error handling
}
```

Then the error check on line 101 is also only for the api.CreateOrUpdateApplication method not also the type assertion.

File: spinnaker/provider.go [hunk 2]
```diff
@@ -1,12 +1,12 @@
 package spinnaker
 
 import (
-   "os"
-   "io/ioutil"
+	"os"
+	"io/ioutil"
```
Comment (pjberry16) [inline spinnaker/provider.go:5]: Pretty sure io/ioutil package is deprecated in favor of just io might be worth switching while making changes.

File: spinnaker/api/util.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package api
+
+import (
+	"fmt"
+	//"net/http"
+
+	//"github.com/mitchellh/mapstructure"
+	//gate "github.com/spinnaker/spin/cmd/gateclient"
```
Comment (pjberry16) [inline spinnaker/api/util.go:8]: Are the commented imports necessary to keep?

File: spinnaker/resource_application.go (hunk 1 above)
Comment (guido9j) [reply spinnaker/resource_application.go]: Eliminated unneeded cast

**Review by pjberry16 — APPROVED** [review]
lgtm

**Pull request merged**